src/pybind/mgr/balancer/module.py

```diff
@@ -485,7 +485,7 @@ def handle_command(self, inbuf, command):
                     should_prune = True
             if should_prune: # some pools were gone, prune
                 self.set_module_option('pool_ids', ','.join(final_ids))
-            return (0, json.dumps(final_names, indent=4, sort_keys=True), '')
+            return (0, json.dumps(sorted(final_names), indent=4, sort_keys=True), '')
```

Comment:
> If `sort_keys` is true (default: False), then the output of dictionaries will be sorted by key.

presumably, the output of `json.dumps()` should already be sorted?

Reply:
In this case it's a list. The sorting only applies to dict keys.